benchmark_v2/framework/benchmark_runner.py

```diff
@@ -274,6 +276,54 @@ def run_benchmark(
                 "config": config,
             }
 
+    def time_generate_batch(
+        self,
+        max_new_tokens: int,
+        gpu_monitor: GPUMonitor | None = None,
+    ) -> tuple[float, list[float], str, GPURawMetrics | None]:
+        if gpu_monitor is not None:
+            gpu_monitor.start()
+        config = GenerationConfig(
+            max_new_tokens=max_new_tokens,
+            use_cuda_graphs=False,
+            eos_token_id=self.tokenizer.eos_token_id,
+            pad_token_id=self.tokenizer.pad_token_id,
+            do_sample=False,
+        )
+        manager = self.model.init_continuous_batching(config)
+        manager.start()
+        first_req_results = []
+        timestamps = []
+        wall_time_0 = time.perf_counter()
+        inputs = self.inputs["input_ids"].tolist()
+        manager.add_requests(inputs, max_new_tokens=max_new_tokens, stream=True)
+        first_req_id = None
+        num_requests = len(inputs)
+        finished_requests = 0
+        while finished_requests < num_requests:
+            timestamps.append(time.perf_counter() - wall_time_0)
```

Comment:
Does this mean that a timestamp is added for each finished request or each generated token? Or none of those?